ravenframework/Optimizers/parentSelectors/parentSelectors.py

```diff
@@ -88,65 +102,50 @@ def tournamentSelection(population,**kwargs):
           variables, list, variable names
     @ Out, newPopulation, xr.DataArray, selected parents,
   """
-  fitness = kwargs['fitness']
-  nParents= kwargs['nParents']
+
+  nParents = kwargs['nParents']
+  nObjVal  = len(kwargs['objVal'])
+  kSelect = kwargs['kSelection']
   pop = population
   popSize = population.values.shape[0]
 
-  if 'rank' in kwargs:
-    # the key rank is used in multi-objective optimization where rank identifies which front the point belongs to
-    rank = kwargs['rank']
-    multiObjectiveRanking = True
-    matrixOperationRaw = np.zeros((popSize,3))
-    matrixOperationRaw[:,0] = np.transpose(np.arange(popSize))
-    matrixOperationRaw[:,1] = np.transpose(fitness.data)
-    matrixOperationRaw[:,2] = np.transpose(rank.data)
-    matrixOperation = np.zeros((popSize,3))
-  else:
-    multiObjectiveRanking = False
-    matrixOperationRaw = np.zeros((popSize,2))
-    matrixOperationRaw[:,0] = np.transpose(np.arange(popSize))
-    matrixOperationRaw[:,1] = np.transpose(fitness.data)
-    matrixOperation = np.zeros((popSize,2))
-
-  indexes = list(np.arange(popSize))
-  indexesShuffled = randomUtils.randomChoice(indexes, size=popSize, replace=False, engine=None)
-
-  if popSize<2*nParents:
-    raise ValueError('In tournamentSelection the number of parents cannot be larger than half of the population size.')
-
-  for idx, val in enumerate(indexesShuffled):
-    matrixOperation[idx,:] = matrixOperationRaw[val,:]
-
-  selectedParent = xr.DataArray(
-    np.zeros((nParents,np.shape(pop)[1])),
-    dims=['chromosome','Gene'],
-    coords={'chromosome':np.arange(nParents),
-            'Gene': kwargs['variables']})
+  selectedParent = xr.DataArray(np.zeros((nParents,np.shape(pop)[1])),
+                                dims=['chromosome','Gene'],
+                                coords={'chromosome':np.arange(nParents),
+                                        'Gene': kwargs['variables']})
 
-  if not multiObjectiveRanking: # single-objective implementation of tournamentSelection
+  if nObjVal == 1: # single-objective Case
```

Comment:
As I mentioned before, please define a variable, such as `self._multiobjetiveMode`, for the check here.